spec/gateways/organisation_users_spec.rb

```diff
@@ -0,0 +1,27 @@
+describe Gateways::OrganisationUsers do
+  subject(:gateway) { described_class.new(organisation: organisation) }
+
+  context 'with users' do
+    let(:organisation) { create(:organisation) }
+    let(:result) do
+      organisation.users
+    end
+
+    before do
+      create_list :user, 3, organisation: organisation
+      create(:user)
+    end
+
+    it 'fetches the users for the organisation' do
+      expect(gateway.fetch).to eq(result)
+    end
+  end
+
+  context 'without users' do
+    let(:organisation) { create(:organisation) }
+
+    it 'returns an empty collection' do
+      expect(gateway.fetch).to eq([])
```

Comment:
We're expecting this gateway to always yield an active record collection. Can it be tested as such please?

I think we're also missing a test to assert that further queries can be run.
I'm unsure on the "rails way" for this, however I think asserting that it's an `ActiveRecord::Collection` should suffice

Reply:
Agree to a point. Heres the behaviour:
```
gateway.fetch
=> []
```
```
gateway.fetch.class
=> User::ActiveRecord_Associations_CollectionProxy
```